modules/shared_data/src/main/scala/org/elpaca_metagraph/shared_data/daemons/fetcher/YouTubeFetcher.scala

```diff
@@ -139,51 +164,27 @@ object YouTubeFetcher {
             }
           }.getOrElse(false)
         }
-
-        _ <- logger.info(s"Found ${latticeUsers.length} Lattice users")
-        _ <- logger.info(s"Found ${filteredLatticeUsers.length} filtered Lattice users")
-
-        dataUpdates <- filteredLatticeUsers.traverse { user =>
-          searchInformation.traverse { searchInfo =>
-            youtubeFetcher.fetchVideosByChannelId(
-              user.youtube.get.channelId,
-              searchInfo.text,
-              searchInfo.minimumDuration,
-              searchInfo.minimumViews,
-              searchInfo.maxPerDay,dataSource.existingWallets.get(user.primaryDagAddress.get).map { wallet =>
-                val addressReward = wallet.addressRewards(searchInfo.text)
-                val latestVideoRewarded = addressReward.videos.sortBy(_.publishedAt)(Ordering[Instant].reverse).head
-                latestVideoRewarded.publishedAt.atZone(zoneOffset).toLocalDateTime
-              }
-            ).map(_.map { video => YouTubeUpdate(
-              user.primaryDagAddress.get,
-              searchInfo.text,
-              video
-            )})
-          }.map(_.flatten)
-        }.map(_.flatten)
-
         newlyLoadedLatticeUsers = latticeUsers.filterNot { user =>
           dataSource.existingWallets.keys.toList.contains(user.primaryDagAddress.get)
         }
 
-        newUpdates <- newlyLoadedLatticeUsers.traverse { user =>
-          searchInformation.traverse { searchInfo =>
-            youtubeFetcher.fetchVideosByChannelId(
-              user.youtube.get.channelId,
-              searchInfo.text,
-              searchInfo.minimumDuration,
-              searchInfo.minimumViews,
-              searchInfo.maxPerDay
-            ).map(_.map { video => YouTubeUpdate(
-              user.primaryDagAddress.get,
-              searchInfo.text,
-              video
-            )})
-          }.map(_.flatten)
+        _ <- logger.info(s"Found ${latticeUsers.length} Lattice users")
+        _ <- logger.info(s"Found ${filteredLatticeUsers.length} filtered Lattice users")
+        _ <- logger.info(s"Found ${newlyLoadedLatticeUsers.length} newly loaded Lattice users")
+
+        updates <- searchInformation.traverse { searchInfo =>
+          for {
+            globalSearchResult <- youtubeFetcher.searchVideos(
+              s"constellation-${searchInfo.text}",
+              searchInfo.maxPerDay,
+              Some(LocalDateTime.now().minusHours(searchInfo.publishedWithinHours))
+            )
+            dataUpdates <- youtubeFetcher.fetchVideoUpdates(filteredLatticeUsers, searchInfo, globalSearchResult)
+            newUpdates <- youtubeFetcher.fetchVideoUpdates(newlyLoadedLatticeUsers, searchInfo, globalSearchResult)
```

Comment:
Sorry, but why are we fetching twice here? I mean filteredLatticeUsers and newlyLoadedLatticeUsers

Reply:
You're right. It doesn't make sense any more. I removed the duplicate call.